crates/gitbutler-branch-actions/src/move_commits.rs

```diff
@@ -46,6 +47,14 @@ pub(crate) fn move_commit(
         bail!("Can not move conflicted commits");
     }
 
+    let is_head_commit = commit_id == source_branch.head;
+    if !is_head_commit {
+        let ids_to_rebase = ctx.repository().l(source_branch.head, LogUntil::Commit(commit_id)).unwrap();
+
+        let _ = cherry_rebase_group(ctx.repository(), source_commit.parent_id(0).unwrap(), &ids_to_rebase, false);
+    }
+
+
```

Comment:
This operation makes sense in principle, but you're missing some things:

1. This needs to be executed only after checking that the commit that's being moved (`source_commit`) is not locked to it's descendants (`ids_to_rebase`). 
In order to do that, you can just rename the `check_source_lock_to_ancestors` function to something more generic like `check_source_lock_to_commits`, and pass the corresponding commits derived from the `ids_to_rebase`.

2. `cherry_rebase_group` returns a commit ID that then needs to be set to the `source_branch.head`.
Almost at the end though, just like we do for the `destination_branch.head` (line 126):

```rs
 let new_destination_head_oid = cherry_rebase_group(
        ctx.repository(),
        destination_branch.head,
        &[source_commit.id()],
        true,
    )?;

    // @Pranav, here's the place you should add the rebase operation.
    // The returned commit ID should be set to the source_branch.head

    // reset the source branch to the parent commit
    // and update the destination branch head
    source_branch.head = source_branch_head_parent.id();
    vb_state.set_branch(source_branch.clone())?;

    destination_branch.head = new_destination_head_oid;
    vb_state.set_branch(destination_branch.clone())?;
```

3. Let's try to avoid to `unwrap`. In order to get the parent commit, you can already use the following variable (line 58)
```rs
    let source_branch_head_parent = source_commit
        .parent(0)
        .context("failed to get parent commit")?;
```